src/App.js

```diff
@@ -1,16 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  // const [dotColors, setDotColors] = useState(#bbb);
+  const [newChatMessages, setNewChatMessages] = useState(chatMessages);
+
+  const createNewMessage = (updatedMessage) => {
+    const newChatData = newChatMessages.map((message) => {
+      if (message.id === updatedMessage.id) {
+        return updatedMessage;
+      } else {
+        return message;
+      }
+    });
+    setNewChatMessages(newChatData);
+  };
```

Comment:
I might consider refactoring this, so that it doesn't update ALL of the message data, just the `liked` property